api_types.ts

```diff
@@ -701,6 +702,11 @@ export interface Sync {
    * The dynamic URL is likely necessary for determining which API resources to fetch.
    */
   dynamicUrl?: string;
+
+  /**
```

Comment:
Another thing to consider here is whether we want to add this for all ExecutionContext types or whether we might want to initially only do it for `PropertyOptionsExecutionContext` by giving it a `PropertyOptionsSync` sub-interface. That'd avoid the need to worry about populating this for pre-existing formulas but still leave the door open to generalize it later

Reply:
Even if we don't support it on all metadata formulas *yet*, I personally think it's better to go with all execution contexts for sync tables, as I think we should get there eventually. 

Actually, I daresay it should go with all execution contexts for formulas & actions too, so maybe even move it higher?

Reply:
Actually, looking at this a little more, why can't `PropertyOptionsMetadataFormula` extend `MetadataFunction` and get this for free?

Reply:
I believe that the reasoning is that, the `MetadataFunction` interface is a bit overloaded (it has a number of parameters that only pertain to specific types of metadata functions, like `search` only being relevant to autocomplete functions). There was also some mention of the positionality of the arguments, although I can't remember off the top of my head the details about that discussion since `MetadataContext` doesn't hold any data about the positional arguments. 

So the thought was to declare `PropertyOptions` as a separate interface to keep things cleaner.

Reply:
More on the positionality: I remember @dweitzman-codaio was talking about a scenario where we applied varargs with the same name i.e. (`cake`, `cake_flavor`, `cake`, `cake_flavor`...) and attempted to do an autocomplete accounting for the aggregate of this info — `MetadataContext` would need a different interface shape to properly handle this scenario. (But I can't remember why this specific example isn't related to not extending `MetadataFunction`)

Reply:
ah, yeah we should make MetadataFunction able to take in varargs

Reply:
Re: why `PropertyOptionsMetadataFunction` doesn't extend `MetadataFunction`, there are two reasons:

1) The existing argument types of `MetadataFunction` wouldn't work well for property autocomplete so we wanted `context` versions of some of the same parameters anyway. For example, `MetadataFunction` has a `search` argument used for parameter autocomplete, but for property autocomplete we use `Object.defineProperty()` trickery to spy on whether the search text is used for implicitly tracking what dependencies the formula has for cache keys. We wouldn't be able to spy whether the `MetadataFunction` `search` function parameter is or isn't accessed, so I didn't use it.

2) More generally, trying to force every kind of metadata function to use the same set of positional arguments and parameters doesn't seem as clean, simple, consistent, and flexible as just putting context into the context object in whatever shape seems nice without regard to to typing of formula parameters.

Today, a get schema or a get connection name metadata function always gets a "search" positional argument which serves no purpose. Because of that, if we wanted to introduce new positional arguments people might end up writing functions like `getSchema: (context, _unusedSearch, _unusedDeprecatedFormulaContext, newerBetterFormulaContextInputThatSupportsVarArgs)`. In terms of typescript annoyances, I assume the only reason `makeMetadataFormula` needs to deserialize `formulaContext` from a string is that it has `ParameterType.String` to be a valid formula parameter and whoever wrote `makeMetadataFormula` was trying to appease typescript by cramming a non-string into a string argument. `ExecutionContext` and its subinterfaces aren't limited to just pack formula parameter types, so it's simpler and more flexible to use than formula arguments.